src/algebra/group_with_zero/defs.lean

```diff
@@ -54,13 +54,24 @@ class no_zero_divisors (M₀ : Type*) [has_mul M₀] [has_zero M₀] : Prop :=
 
 export no_zero_divisors (eq_zero_or_eq_zero_of_mul_eq_zero)
 
+/-- A type `S₀` is a "semigroup with zero” if it is a semigroup with zero element, and `0` is left
+and right absorbing. -/
+@[protect_proj] class semigroup_with_zero (S₀ : Type*) extends semigroup S₀, mul_zero_class S₀.
+
+/- By defining this _after_ `semigroup_with_zero`, we ensure that searches for `mul_zero_class` find
```

Comment:
Do we really have to rely on the order in the file? Would it be possible to achieve the same results by setting an priority?

Reply:
Setting priority 99 on `semigroup_with_zero.to_mul_zero_class` would work too, but I'd argue it makes things more complex as that would cause other files that try to be lower priority with `99` to have to switch to 98. We don't have a particularly good method of bookkeeping priorities right now (the docs don't show it), so my inclination is to avoid them  where possible.

Reply:
Fair enough. I admit neither solution is ideal.